src/frontend/js/utils/api/lms/joanie.ts

```diff
@@ -0,0 +1,85 @@
+import JoanieApi from 'utils/api/joanie';
+import { AuthenticationBackend, LMSBackend } from '../../../types/commonDataProps';
+import { APILms } from '../../../types/api';
+
+import { Maybe, Nullable } from '../../../types/utils';
+import { User } from '../../../types/User';
+import * as Joanie from '../../../types/Joanie';
+import { Enrollment } from '../../../types/Joanie';
+
+const JoanieEnrollmentApiInterface = (
+  APIConf: AuthenticationBackend | LMSBackend,
+): APILms['enrollment'] => {
+  const joanieAPI: Joanie.API = JoanieApi();
+
+  const extractCourseRunIdFromUrl = (url: string): Maybe<Nullable<string>> => {
+    const matches = url.match((APIConf as LMSBackend).course_regexp);
+    return matches && matches[2] ? matches[2] : null;
+  };
+  return {
+    get(url: string) {
+      return new Promise((resolve) => {
+        const courseRunId = extractCourseRunIdFromUrl(url);
+        joanieAPI.user.enrollments
+          .get({ id: undefined, course_run: courseRunId as string })
```

Comment:
You should not need to provide an enrollmentId.
```suggestion
          .get({ course_run: courseRunId as string })
```